.github/workflows/ci.yml

```
@@ -21,8 +21,11 @@ jobs:
       # remove occasionally problematic repositories we don't use anyway
       run: |
         sudo rm -f /etc/apt/sources.list.d/dotnetdev.list /etc/apt/sources.list.d/microsoft-prod.list
-        sudo apt update
-        sudo apt install texlive-base texlive-generic-recommended texlive-fonts-recommended texlive-latex-base texlive-latex-recommended texlive-latex-extra dvipng dvidvi
+        sudo gem install apt-spy2
+        sudo apt-spy2 check
+        sudo apt-spy2 fix
```

```suggestion
        sudo apt-spy2 fix --commit
```

these lines are redundant - `check` apparently only exists for convenience, to output the best mirror. `fix` does the same, but can also apply that change to `/etc/apt/sources.list`.

To actually apply the change, you need to provide the `--commit` argument:
```bash
florian@florian-vm-lubuntu1910:~$ sudo apt-spy2 fix
[sudo] password for florian: 
The closest mirror is: http://ftp.rrzn.uni-hannover.de/pub/mirror/linux/ubuntu/
Run with --commit to adjust /etc/apt/sources.list
```

The same applies to all other usages in this changeset.

---

So for the previous runs, `apt-spy2` didn't actually apply a change, but we were indeed still running on the original addresses? Hmm.

---

Yes, you can see apt fetching from `azure.archive.ubuntu.com` just below the log line i linked.

---

Sure. My point was just that the temporary fix was not needed from the get go then.
I have applied your suggestion, thanks! :) But think I will advise to let this PR lie around for some time, to see if the issue persists and only merge it if it does.